dasharo-stability/capsule-update.robot

```diff
@@ -0,0 +1,258 @@
+...                     Turn Off Active ME
+Suite Teardown      Run Keywords
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+CUP001.001 Capsule Update With Wrong Keys
+    [Documentation]    This test aims to verify if the DUT will reject flashed using Capsule Update with capsule signed with invalid certificates.
+    Power On
+    ${boot_menu}=    Enter Boot Menu Tianocore And Return Construction
+    Enter Submenu From Snapshot    ${boot_menu}    UEFI Shell
+    Read From Terminal Until    Shell>
+
+    ${out}=    Execute UEFI Shell Command    smbiosview -t 0
+    ${original_bios_version}=    Extract BIOS Version    ${out}
+    Log To Console    \nBefore update ${original_bios_version}
+
+    ${out}=    Start Update Process    wrong_cert.cap
+    Should Not Contain    ${out}    (The platform will automatically reboot and disable Firmware Update Mode
+    Should Not Contain    ${out}    failed to query capsule capability
+    Should Contain    ${out}    CapsuleApp: creating capsule descriptors at
+    Should Contain    ${out}    :\\capsule_testing\\>
+
+    Set DUT Response Timeout    3m
+    ${out}=    Execute UEFI Shell Command    reset
+    Should Not Contain    ${out}    (The platform will automatically reboot and disable Firmware Update Mode
+
+    ${boot_menu}=    Enter Boot Menu Tianocore And Return Construction
+    Enter Submenu From Snapshot    ${boot_menu}    UEFI Shell
+    Read From Terminal Until    Shell>
+    ${out}=    Execute UEFI Shell Command    smbiosview -t 0
+    ${updated_bios_version}=    Extract BIOS Version    ${out}
+    Log To Console    \nAfter update ${updated_bios_version}
+
+    Enter Capsule Testing Folder
+    ${out}=    Execute UEFI Shell Command    CapsuleApp.efi -S
+    IF    '${CONFIG}' == 'qemu'
+        Should Contain    ${out}    Capsule Status: Security Violation
+    ELSE
+        Should Contain    ${out}    Capsule Status: Not Ready
+    END
+
+CUP002.001 Capsule Update With Wrong GUID
+    [Documentation]    This test aims to verify if the DUT will reject flashed using Capsule Update with capsule with invalid GUID.
+    Power On
+    ${boot_menu}=    Enter Boot Menu Tianocore And Return Construction
+    Enter Submenu From Snapshot    ${boot_menu}    UEFI Shell
+    Read From Terminal Until    Shell>
+
+    ${out}=    Execute UEFI Shell Command    smbiosview -t 0
+    ${original_bios_version}=    Extract BIOS Version    ${out}
+    Log To Console    \nBefore update ${original_bios_version}
+
+    ${out}=    Start Update Process    invalid_guid.cap
+    Should Not Contain    ${out}    (The platform will automatically reboot and disable Firmware Update Mode
+    Should Not Contain    ${out}    failed to query capsule capability
+    Should Contain    ${out}    CapsuleApp: creating capsule descriptors at
+    Should Contain    ${out}    :\\capsule_testing\\>
+
+    Set DUT Response Timeout    3m
+    ${out}=    Execute UEFI Shell Command    reset
```

Comment:
This looked non-intuitive (I was expecting "execute command" to read until the next prompt, which wouldn't show up for `reset`) so I've looked how `Execute UEFI Shell Command` is implemented. It returns output from 1 second after sending enter key, which isn't elegant and probably doesn't work properly for commands with longer output anyway. Could you create an issue to fix the implementation of `Execute UEFI Shell Command` and link it here?

Also, this is the reason why `Set DUT Response Timeout` didn't have any effect for this line. As it doesn't impact the reset, please move it to the next set of commands, just before `Enter Boot Menu...`, same for all other test cases.

Reply:
Execute UEFI Shell Command has its internal timeout. Thats why global setting didnt work. I have changed tests to make use of it: f951b1304e0010cb21ff688f02942344e7d11d8b

However the output of this command gathers only 1 sec of logs still. So issue was created: https://github.com/Dasharo/open-source-firmware-validation/issues/490